packages/core/src/index.ts

```diff
@@ -0,0 +1,190 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```

Comment:
I have added most of these interfaces already and I dont this needs to be rebased and only newer types should be added